### PR TITLE
New functionalities PR 2

### PR DESCRIPTION
**Pull request by ad-47**
Added new RPC and test cases for logging single fields (eg: LogMetric, LogAttribute, LogHyperparameter etc.)
Added check for protecting the duplicate entry of attribute, datasets, hyperparameter, etc.
Silent duplicate tag logging

### CODE REVIEW

File: src/main/java/com/mitdbg/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 1]
```diff
@@ -201,6 +201,23 @@ public Boolean deleteExperimentRun(String experimentRunId) {
     }
   }
 
+  @Override
+  @Transactional
+  public ExperimentRun getExperimentRun(String experimentRunId)
+      throws InvalidProtocolBufferException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Transaction transaction = session.beginTransaction();
```
Comment (ravishetye) [inline src/main/java/com/mitdbg/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: is it common to wrap  read only operations in transaction in Hibernate?

Comment (ravishetye) [reply src/main/java/com/mitdbg/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: @ad-47  : did you follow up on this?

**Comment by ad-47** [comment]
@ravishetye, This PR looks good to me and for remaining comment I have raised the Issue so if you are ok with it please merge this subsequence PR.

File: src/main/java/com/mitdbg/modeldb/experimentRun/ExperimentRunDAORdbImpl.java (hunk 1 above)
Comment (ad-47) [reply src/main/java/com/mitdbg/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: Created the new ticket for this. we will cover it in separate PR.

**Pull request merged**